pkg/scaling/resolver/aws_secretmanager_handler.go

```diff
@@ -0,0 +1,120 @@
+	}
+	if versionStage != "" {
+		input.VersionStage = aws.String(versionStage)
+	}
+	result, err := ash.session.GetSecretValue(ctx, input)
+	if err != nil {
+		logger.Error(err, "Error getting credentials")
+		return "", err
+	}
+	return *result.SecretString, nil
+}
+
+func (ash *AwsSecretManagerHandler) Initialize(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, secretsLister corev1listers.SecretLister, podTemplateSpec *corev1.PodTemplateSpec) error {
+	config, err := ash.getcredentials(ctx, client, logger, triggerNamespace, secretsLister, podTemplateSpec)
+	if err != nil {
+		logger.Error(err, "Error getting credentials")
+		return err
+	}
+
+	logger.Info("Config value", "config", config)
+
+	if ash.secretManager.Cloud != nil {
+		if ash.secretManager.Cloud.Region != "" {
+			config.Region = ash.secretManager.Cloud.Region
+		}
+	}
+
+	ash.session = secretsmanager.NewFromConfig(*config)
+	return nil
+}
+
+func (ash *AwsSecretManagerHandler) getcredentials(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, secretsLister corev1listers.SecretLister, podTemplateSpec *corev1.PodTemplateSpec) (*aws.Config, error) {
+	config := &aws.Config{}
+
+	podIdentity := ash.secretManager.PodIdentity
+	if podIdentity == nil {
+		podIdentity = &kedav1alpha1.AuthPodIdentity{}
+	}
+
+	switch podIdentity.Provider {
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		accessKeyID := resolveAuthSecret(ctx, client, logger, ash.secretManager.Credentials.AccessKey.ValueFrom.SecretKeyRef.Name, triggerNamespace, ash.secretManager.Credentials.AccessKey.ValueFrom.SecretKeyRef.Key, secretsLister)
+		accessSecretKey := resolveAuthSecret(ctx, client, logger, ash.secretManager.Credentials.AccessSecretKey.ValueFrom.SecretKeyRef.Name, triggerNamespace, ash.secretManager.Credentials.AccessSecretKey.ValueFrom.SecretKeyRef.Key, secretsLister)
+		if accessKeyID == "" || accessSecretKey == "" {
+			return nil, fmt.Errorf("AccessKeyID and AccessSecretKey are expected when not using a pod identity provider")
+		}
+		config.Credentials = credentials.NewStaticCredentialsProvider(
+			accessKeyID,
+			accessSecretKey,
+			"",
+		)
+		return config, nil
+
+	case kedav1alpha1.PodIdentityProviderAwsEKS:
+		awsRoleArn, err := ash.getRoleArnAwsEKS(ctx, client, logger, triggerNamespace, podTemplateSpec)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving role arn for AwsEKS pod identity: %w", err)
+		}
+		config.Credentials = stscreds.NewAssumeRoleProvider(sts.NewFromConfig(*config), awsRoleArn)
+		return config, nil
+	case kedav1alpha1.PodIdentityProviderAwsEKSWorkload:
+		awsRoleArn, err := ash.getRoleArnAwsEKS(ctx, client, logger, triggerNamespace, podTemplateSpec)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving role arn for AwsEKS workload identity: %w", err)
+		}
+		config.Credentials = stscreds.NewAssumeRoleProvider(sts.NewFromConfig(*config), awsRoleArn)
+		return config, nil
```

Comment:
same as above

Reply:
I'm aware of that. it a dummy code.
I'm yet to start the the actual logic